### PR TITLE
First draft User structure

### LINKED ISSUE #40: Re-enable "noImplicitAny" and "strictNullChecks" in tsconfig.back.json

### PR DESCRIPTION
**Pull request by Hgallien**
- [x]  remove base-entity
- [x] refacto user around id instead of named
- [x] add friendlist
- [x] create multiple http route
- [x] add avatar jpeg

Closes #40

### CODE REVIEW

**Review by DoomDuck — CHANGES_REQUESTED** [review]
You don't follow the convention the naming convention:
> There is PR on this #54 that needs review

Channel related code should not be in this PR (please move it to another PR)
There is an inconsistency with plurals (I would personally go without plural there)
You have `databaseFiles.controller.ts` and `databaseFile.entity.ts`
It is also hard to follow with implicit return types I think it would be nice to [block this in the tsconfig](https://www.typescriptlang.org/tsconfig#noImplicitReturns)

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
```
Comment (DoomDuck) [inline tsconfig.json:1]: This shouldn't be here :eyes:

**Review by DoomDuck — CHANGES_REQUESTED** [review]
missing the new `nest-cli.json`
And previous concerns have not been addressed

File: src/back/user/databaseFiles.controller.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    const stream = Readable.from(file.data);
+    response.set({
+      "Content-Disposition": `inline; filename="${file.filename}"`,
+      "Content-Type": "image",
+    });
+
+    return new StreamableFile(stream);
+  }
+}
```
[the remark this answers is not in the document]
Comment (Hgallien) [reply src/back/user/databaseFiles.controller.ts]: UseInterceptors is useless here, the decorator is use on the class but the one fonction in it doesn't use it. I've removed it and it work

File: package.json [hunk 3]
```diff
@@ -56,6 +56,7 @@
 		"@types/jest": "^28.1.0",
 		"@types/node": "^16.0.0",
 		"@types/three": "^0.138.0",
+		"class-transformer": "^0.5.1",
```
[the remark this answers is not in the document]
Comment (DoomDuck) [reply package.json:60]: Shouldn't we remove this if it is not needed

File: src/back/user/game.entity.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { Id } from "../customType";
+import {
+  Entity,
+  Column,
+  PrimaryGeneratedColumn,
+  CreateDateColumn,
+} from "typeorm";
+@Entity("game")
+export class Gane {
```
Comment (DoomDuck) [inline src/back/user/game.entity.ts:9]: I think this is a typo

File: src/back/user/game.entity.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  playerOne: string;
+
+  @Column({ type: "varchar", length: 100, nullable: false })
+  playerTwo: string;
+
+  @Column("int")
+  playerOneScore: Id;
+
+  @Column("int")
+  playerTwoScore: Id;
```
Comment (DoomDuck) [inline src/back/user/game.entity.ts:22]: Those should be `number` not `Id`

File: src/back/user/game.entity.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+} from "typeorm";
+@Entity("game")
+export class Gane {
+  @PrimaryGeneratedColumn()
+  id: Id;
+  @Column({ type: "varchar", length: 100, nullable: false })
+  playerOne: string;
+
+  @Column({ type: "varchar", length: 100, nullable: false })
+  playerTwo: string;
```
Comment (DoomDuck) [inline src/back/user/game.entity.ts:16]: Those should likely be `Id`s

File: src/back/user/requestWithUser.interface.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+import { UserDto } from "./user.dto";
+
+import { Id } from "../customType";
+interface RequestWithUser extends Request {
+  //might be needed later
+  // user: UserDto;
+  id: Id;
+}
+
+export default RequestWithUser;
```
Comment (DoomDuck) [inline src/back/user/requestWithUser.interface.ts:11]: This doesn't seem to useful maybe it shouldn't be in the PR

**Review by DoomDuck — CHANGES_REQUESTED** [review]
Game parts should likely not be part of this PR

Almost there :slightly_smiling_face:

**Review by DoomDuck — DISMISSED** [review]
Welldone :+1:

**Review by DoomDuck — APPROVED** [review]
